src/Control/SAMLController.php

```diff
@@ -154,12 +154,14 @@ public function metadata()
     protected function getRedirect()
     {
         // Absolute redirection URLs may cause spoofing
-        if (Session::get('BackURL') && Director::is_site_url(Session::get('BackURL'))) {
-            return $this->redirect(Session::get('BackURL'));
+        if ($this->getRequest()->getSession()->get('BackURL') &&
+            Director::is_site_url($this->getRequest()->getSession()->get('BackURL'))) {
```

Comment:
If dropping a line I think it's nice if you take the `&&` operator with it, so it doesn't look like a statement on its own

Reply:
I agree. Fixed this and the one a bit further down below 👍